src/connection/host_lookup.c

```diff
@@ -0,0 +1,635 @@
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+/*
+ * host_lookup.c
+ */
+
+#ident "$Id$"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include <string.h>
+#include <limits.h>
+#include <errno.h>
+#include <unordered_map>
+
+#if defined (WINDOWS)
+#include <winsock2.h>
+#include <ws2tcpip.h>
+#else
+#include <arpa/inet.h>
+#include <netdb.h>
+#endif
+
+#include "system_parameter.h"
+#include "environment_variable.h"
+#include "message_catalog.h"
+
+#define HOSTNAME_LEN                 (256)
+#define MAX_NUM_HOSTS                (256)
+#define LINE_BUF_SIZE                (512)
+#define IPADDR_LEN                   (17)
+#define NUM_IPADDR_DOT               (3)
+#define IPv4_ADDR_LEN                (4)
+
+#define NUM_DIGIT(VAL)              (size_t)(log10 (VAL) + 1)
+#define FREE_MEM(PTR)           \
+        do {                    \
+          if (PTR) {            \
+            free(PTR);          \
+            PTR = 0;    \
+          }                     \
+        } while (0)
+
+#define FREE_HOSTENT_MEM(PTR)        \
+        do {                    \
+          if (PTR) {            \
+            FREE_MEM (PTR->h_aliases[0]);       \
+            FREE_MEM (PTR->h_addr_list[0]);     \
+            FREE_MEM (PTR->h_addr_list);                \
+            FREE_MEM (PTR->h_aliases);          \
+            FREE_MEM (PTR->h_name);             \
+            FREE_MEM (PTR);                     \
+           }                                    \
+        }while(0)
+
+typedef enum
+{
+  HOSTNAME_TO_IPADDR = 0,
+  IPADDR_TO_HOSTNAME = 1,
+} LOOKUP_TYPE;
+
+typedef enum
+{
+  INSERT_IPADDR = 0,
+  INSERT_HOSTNAME = 1,
+} HOSTENT_INSERT_TYPE;
+
+typedef enum
+{
+  USE_GLIBC_HOSTS = 0,
+  USE_USER_DEFINED_HOSTS = 1,
+} HOST_LOOKUP_TYPE;
+
+typedef enum
+{
+  LOAD_FAIL = -1,
+  LOAD_INIT,
+  LOAD_SUCCESS,
+} HOSTS_CONF_LOAD_STATUS;
+
+static const char *user_hosts_File = "hosts.conf";
+static struct hostent *hostent_Cache[MAX_NUM_HOSTS];
+
+static int hosts_conf_file_Load = LOAD_INIT;
+
+// *INDENT-OFF*
+static std::unordered_map <std::string, int> user_host_Map;
+// *INDENT-ON*
+
+static struct hostent *hostent_alloc (char *ipaddr, char *hostname);
+static bool ip_format_check (char *ip_addr);
+static int load_hosts_file ();
+static struct hostent *host_lookup_internal (const char *hostname, struct sockaddr *saddr, LOOKUP_TYPE lookup_type);
+
+/*
+ * hostent_alloc () - Allocate memory hostent structure.
+ * 
+ * return   : the hostent pointer.
+ *
+ */
+static struct hostent *
+hostent_alloc (char *ipaddr, char *hostname)
+{
+  struct hostent *hp;
+  char addr_trans_bi_buf[IPADDR_LEN];
```

Comment:
If it is a buffer that stores binary data, it is butter to set the size to 4.

Reply:
Right. 
I tested and its size of 4.

Reply:
For your information, 
if you look at the Linux man page (man inet_pton), it is sized as follows.

`unsigned char buf[sizeof(struct in6_addr)];`

Reply:
@ctshim 
I just use sizeof(struct in_addr) instead of size 4.

> "The address is converted to a struct in_addr and copied to dst, which must be sizeof(struct in_addr) (4) bytes (32 bits) long" in the inet_pton page of linux man page

I read this and then apply it with example code.
And I cannot apply your suggestion 
`sockaddr_in addr{ AF_INET, htons(PORT) };
inet_pton(AF_INET, "127.0.0.1", &(addr.sin_addr.s_addr));` through the teams chatting because there is no port ID. and apply example in the man page. 

Could you review the fixed code?